main.go

```diff
@@ -443,31 +444,138 @@ func run(state overseer.State) {
+		ScanEntireChunk:          *scanEntireChunk,
+		JobReportWriter:          jobReportWriter,
+		AllowVerificationOverlap: *allowVerificationOverlap,
+		ParsedResults:            parsedResults,
+		Printer:                  printer,
+	}
+
+	if *compareDetectionStrategies {
+		err := compareScans(ctx, scanConfig)
+		if err != nil {
+			logFatal(err, "error comparing span calculators")
+		}
+		return
+	}
+
+	metrics, err := runSingleScan(ctx, scanConfig, *scanEntireChunk)
+	if err != nil {
+		logFatal(err, "error running scan")
+	}
+
+	// Print results.
+	logger.Info("finished scanning",
+		"chunks", metrics.ChunksScanned,
+		"bytes", metrics.BytesScanned,
+		"verified_secrets", metrics.VerifiedSecretsFound,
+		"unverified_secrets", metrics.UnverifiedSecretsFound,
+		"scan_duration", metrics.ScanDuration.String(),
+		"trufflehog_version", version.BuildVersion,
+	)
+
+	if metrics.hasFoundResults && *fail {
+		logger.V(2).Info("exiting with code 183 because results were found")
+		os.Exit(183)
+	}
+
+}
+
+type scanConfig struct {
+	Command                  string
+	Concurrency              int
+	Decoders                 []decoders.Decoder
+	Conf                     *config.Config
+	IncludeFilter            func(detectors.Detector) bool
+	ExcludeFilter            func(detectors.Detector) bool
+	EndpointCustomizer       func(detectors.Detector) bool
+	NoVerification           bool
+	PrintAvgDetectorTime     bool
+	FilterUnverified         bool
+	FilterEntropy            float64
+	ScanEntireChunk          bool
+	JobReportWriter          io.WriteCloser
+	AllowVerificationOverlap bool
+	ParsedResults            map[string]struct{}
+	Printer                  engine.Printer
+}
+
+func compareScans(ctx context.Context, cfg scanConfig) error {
+	// Run scan with max-length span calculator.
+	maxLengthMetrics, err := runSingleScan(ctx, cfg, false)
+	if err != nil {
+		return fmt.Errorf("error running scan with custom span calculator: %v", err)
+	}
+
+	// Run scan with entire chunk span calculator.
+	entireMetrics, err := runSingleScan(ctx, cfg, true)
+	if err != nil {
+		return fmt.Errorf("error running scan with entire chunk span calculator: %v", err)
+	}
+
+	logComparisonMetrics(maxLengthMetrics.Metrics, entireMetrics.Metrics)
+
+	return nil
+}
+
+func logComparisonMetrics(customMetrics, entireMetrics engine.Metrics) {
+	fmt.Printf("Comparison of scan results: \n")
+	fmt.Printf("Custom span - Chunks: %d, Bytes: %d, Verified Secrets: %d, Unverified Secrets: %d, Duration: %s\n",
+		customMetrics.ChunksScanned, customMetrics.BytesScanned, customMetrics.VerifiedSecretsFound, customMetrics.UnverifiedSecretsFound, customMetrics.ScanDuration.String())
+	fmt.Printf("Entire chunk - Chunks: %d, Bytes: %d, Verified Secrets: %d, Unverified Secrets: %d, Duration: %s",
+		entireMetrics.ChunksScanned, entireMetrics.BytesScanned, entireMetrics.VerifiedSecretsFound, entireMetrics.UnverifiedSecretsFound, entireMetrics.ScanDuration.String())
+}
```

Comment:
I think it would be good to emit an error if the secret counts don't match so we can surface and exit code. This would make a good future smoke test case.

Reply:
Done.